Hazel/src/Hazel/ImGui/ImGuiConsoleSink.h

```diff
@@ -0,0 +1,62 @@
+#include "spdlog/sinks/base_sink.h"
+#include "ImGuiConsole.h"
+
+namespace Hazel {
+
+	template<class Mutex>
+	class ImGuiConsoleSink : public spdlog::sinks::base_sink<std::mutex>
+	{
+	public:
+		explicit ImGuiConsoleSink(bool forceFlush = false, uint8_t bufferCapacity = 10)
+			: m_MessageBufferCapacity(forceFlush ? 1 : bufferCapacity), m_MessageBuffer(std::vector<Ref<ImGuiConsole::Message>>(forceFlush ? 1 : bufferCapacity))
+		{
+		}
+		ImGuiConsoleSink(const ImGuiConsoleSink&) = delete;
+		ImGuiConsoleSink& operator=(const ImGuiConsoleSink&) = delete;
+		virtual ~ImGuiConsoleSink() = default;
+	protected:
+		void sink_it_(const spdlog::details::log_msg& msg) override
+		{
+			fmt::memory_buffer formatted;
+			sink::formatter_->format(msg, formatted);
+
+			*(m_MessageBuffer.begin() + m_MessagesBuffered) = std::make_shared<ImGuiConsole::Message>(fmt::to_string(formatted), GetMessageLevel(msg.level));
+			if (++m_MessagesBuffered == m_MessageBufferCapacity)
+				flush_();
+		}
+
+		void flush_() override
```

Comment:
This seems like its going to keep the old messages around even after they are unused